src/wujihandpy/__init__.py

```
@@ -1,6 +1,56 @@
+        # performs 6 fresh SDO reads (see wrapper.hpp:329). Until C++
+        # exposes a cached variant this re-reads values that the C++
+        # constructor already fetched. Acceptable on modern firmware
+        # (~300 ms, gated by version check) but worth a future fix.
+        sn = ""
+        if raw_version is not None:
+            byte_major = raw_version & 0xFF
+            byte_minor = (raw_version >> 8) & 0xFF
+            if (byte_major, byte_minor) >= (1, 1):
+                try:
+                    sn = self.get_product_sn() or ""
+                except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)